app/src/main/java/net/gsantner/markor/format/asciidoc/AsciidocSyntaxHighlighter.java

```diff
@@ -0,0 +1,330 @@
+        // adapted for Markdown
+        // maybe, needs to be adapted for AsciiDoc?
+        // but not in the current Pull Request
+        // createSmallBlueLinkSpans();
+
+        //TODO: doesn't yet work, but it is called
+        if (_highlightBiggerHeadings) {
+            createSpanForMatches(HEADING_SIMPLE,
+                    new WrMarkdownHeaderSpanCreator(_spannable, AD_COLOR_HEADING, _textSize));
+        } else {
+            createColorSpanForMatches(HEADING_SIMPLE, AD_COLOR_HEADING);
+        }
+
+        if (_highlightCodeChangeFont) {
+            createMonospaceSpanForMatches(MONOSPACE);
+            createMonospaceSpanForMatches(BLOCK_DELIMITED_LISTING);
+            createMonospaceSpanForMatches(BLOCK_DELIMITED_LITERAL);
+            createMonospaceSpanForMatches(LIST_UNORDERED);
+            createMonospaceSpanForMatches(LIST_ORDERED);
+            createMonospaceSpanForMatches(ATTRIBUTE_DEFINITION);
+            createMonospaceSpanForMatches(ADMONITION);
+        }
+
+        createStyleSpanForMatches(BOLD, Typeface.BOLD);
+        createStyleSpanForMatches(ITALICS, Typeface.ITALIC);
+
+        createColorSpanForMatches(LINK_PATTERN, AD_COLOR_LINK);
+        createColorSpanForMatches(XREF_PATTERN, AD_COLOR_LINK);
+        createColorSpanForMatches(IMAGE_PATTERN, AD_COLOR_LINK);
+        createColorSpanForMatches(INCLUDE_PATTERN, AD_COLOR_LINK);
+
+        createColorSpanForMatches(LIST_UNORDERED, AD_COLOR_LIST);
+        createColorSpanForMatches(LIST_ORDERED, AD_COLOR_LIST);
+        createColorSpanForMatches(LIST_DESCRIPTION, AD_COLOR_LIST_DESCRIPTION);
+        //TODO: test for interfernce with other role like underline, line-through => interference
+        createColorSpanForMatches(ROLE_GENERAL, AD_COLOR_ROLE_GENERAL);
+//        createColorSpanForMatches(ATTRIBUTE_DEFINITION, AD_COLOR_ATTRIBUTE);
+
+        createStyleSpanForMatches(ADMONITION, Typeface.BOLD);
+        createColorSpanForMatches(ADMONITION, AD_COLOR_ADMONITION);
+
+        createColorBackgroundSpan(SQUAREBRACKETS, _isDarkMode ? AD_COLORBACK_DARK_SQUAREBRACKETS : AD_COLORBACK_LIGHT_SQUAREBRACKETS);
+        createColorBackgroundSpan(BLOCKTITLE, _isDarkMode ? AD_COLORBACK_DARK_BLOCKTITLE : AD_COLORBACK_LIGHT_BLOCKTITLE);
+
+
+        if (_highlightLineEnding) {
+            createColorBackgroundSpan(HARD_LINE_BREAK, _isDarkMode ? AD_COLORBACK_DARK_MONOSPACE : AD_COLORBACK_LIGHT_MONOSPACE);
+//            //test markdown original pattern, same issues, when content is created in windows
+//            createColorBackgroundSpan(DOUBLESPACE_LINE_ENDING, AD_COLOR_CODEBLOCK);
+        }
+
+        if (_highlightCodeBlock) {
+            createColorBackgroundSpan(MONOSPACE, _isDarkMode ? AD_COLORBACK_DARK_MONOSPACE : AD_COLORBACK_LIGHT_MONOSPACE);
+//            createColorSpanForMatches(CODE, AD_COLOR_TEXT_ON_COLORBACKGROUND);
+            createColorBackgroundSpan(BLOCK_DELIMITED_LISTING, _isDarkMode ? AD_COLORBACK_DARK_MONOSPACE
```

Comment:
I am unsure how well our highlighting system will work for blocks of text which spans multiple lines. It _should_ work, but if you see weird issues please let me know.

Also note that using the `create*` functions with regex is not strictly necessary. You can just call `addSpanGroup` to add a highlighting span if would prefer to use code to detect the region of interest.

(I will attempt to document this better soon)

Reply:
multiline code highlighting works fine so far. Of couse only one background color at the same time is assigned.

@harshad1  - could you explain what is the difference between create* and just addSpanGroup?
I just had a look on the existing syntax highlighting and tried to use the same functions.
But I also could adapt this later.

I am nearly ready for a first "release", still test it a bit on my "real" Smartphone and not only in the emulator and adapt color combinations a bit.

And there is a strange effect: I big file, which I use to test on my smartphone, is not "highlighted" in the Android Emulator.

Also, I did not find, if it is possible, to assign bold and italic at the same time. regex works fine, but bold is not applied in combination with italic. Maybe this is a question or calling order. But these are minor issues and could be solved by me later.

Reply:
All the create* functions basically just wrap the `addSpanGroup` function 

https://github.com/gsantner/markor/blob/master/app/src/main/java/net/gsantner/markor/frontend/textview/SyntaxHighlighterBase.java#L376

Basically all of our highlighting is done with regex patterns so we use the `create*` wrappers.

For your case all you need is
 
`createSpanForMatches(pattern, new HighlightSpan().setBold(true).setItalics(true), groupsToMatch);`

- The [HighlightSpan class](https://github.com/gsantner/markor/blob/master/app/src/main/java/net/gsantner/markor/frontend/textview/SyntaxHighlighterBase.java#L455) is a helper I made to make basic spans easy to apply.
- `pattern` will be your regex pattern
- `groupsToMatch` will be the capture groups in your regex pattern which you want the highlighting applied to.

If you want to use `addSpanGroup` directly just call it with instances of your span and start, end regions. You can use my HighlightSpan class (I would recommend it), but you don't have to

Reply:
I didn't find, how to set MONOSPACE, using createSpanForMatches

        // TODO: How to use setTypeface to set MONOSPACE?
        // setTypeface(Typeface.MONOSPACE) - invalid
        // setTypeface(0) - doesn't work
        createSpanForMatches(ADMONITION, new HighlightSpan().setBold(true).setForeColor(
                AD_FORECOLOR_ADMONITION));
        createMonospaceSpanForMatches(ADMONITION);

Reply:
Your code sample should work correctly.

It may be possible to add typeface functionality to HighlightSpan. Typeface is a property of the TextPaint which the HighlightSpan class manipulates - https://github.com/gsantner/markor/blob/b918ba45c1d926fd7ea074ffa1d708731cdb6b56/app/src/main/java/net/gsantner/markor/frontend/textview/SyntaxHighlighterBase.java#L508


i.e. add a monospace setters and getters and set it in updateDrawState. Reducing the total number of spans will help improve performance.

Reply:
generally said, dynamic background color font size and font families are the most performance relevant things. So especially those things you would do for code blocks (thats why its a opt-in setting at other formats)

So suggestion is to keep it at minimum.

Reply:
On my smartphone, it all works well and performs well, even for large files.
In Android Studio's Android emulator, it doesn't work with larger files. That may have something to do with it.

After all, I didn't put that much energy into it to just turn it all off then, but because Syntax Highlighting is supposed to add value.

I could imagine extending the options to turn on and off different syntax highlighting. At the moment, there are already some. However, I wanted to implement the basic functions for AsciiDoc Format first and do such fine-tuning (maybe also based on feedback from other AsciiDoc users) later. After all, one would have to use more settings for this to differentiate even more: For example:

- color highlighting of AsciiDoc blocks.
- ...

For this you would have to agree where these settings should appear, because on the one hand they are not relevant for other formats, on the other hand such settings should nevertheless be made in one place.

The goal of this PR was a minimally invasive intervention, so that the functionality of the existing formats does not change. And that already starts with such questions, where and how to make additional settings for syntax highlighting.

There are a few other things I can't solve at the moment, like increasing the font size of headers. But these are less important things that can be addressed later.

From my point of view, a review is possible, and I think the implementation of the AsciiDoc format can be adopted.

Reply:
For code block and automatic monospace font there are respective two global options in the editor settings sections. So for that at last nothing additional required on settings, other than a if(enabled/disabled) { createFontHighlight }

Reply:
OK, I will change the usage of these options a bit.